tracing/3702/runtime/evm_tracer/src/lib.rs

```diff
@@ -25,14 +25,29 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 pub mod tracer {
+	use ethereum_types::H256;
 	use evm_tracing_events::{EvmEvent, GasometerEvent, RuntimeEvent, StepEventFilter};
-	use parity_scale_codec::Encode;
+	use parity_scale_codec::{Decode, Encode};
 
 	use evm::tracing::{using as evm_using, EventListener as EvmListener};
 	use evm_gasometer::tracing::{using as gasometer_using, EventListener as GasometerListener};
 	use evm_runtime::tracing::{using as runtime_using, EventListener as RuntimeListener};
+	use sp_runtime::DispatchError;
 	use sp_std::{cell::RefCell, rc::Rc};
 
+	/// The current EthereumXcmTransaction trace status.
+	#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq)]
+	pub enum EthereumTracingStatus {
+		/// A full block trace.
+		Block,
+		/// A single transaction.
+		Transaction(H256),
+		/// Exit signal.
+		TransactionExited,
+	}
+
+	environmental::environmental!(ETHEREUM_TRACING_STATUS: EthereumTracingStatus);
```

Comment:
I am new to this crate, just double-checking. We are using `environmental` to define a global variable that keeps track of what we want to trace, then we check for `EthereumTracer::status` and we trace accordingly.

We rely on this because there is no easy way to pass such parameter down to `CallDispatcher` given trait constraints, right?